### PR TITLE
Use GitHub Actions instead of Travis CI

### PR DESCRIPTION
**Pull request by sagudev**
Average build times are better.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  CARGO_TERM_COLOR: always
+  SHELL: /bin/bash
+
+jobs:
+  Build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest]
```
Comment (jdm) [inline .github/workflows/rust.yml:15]: Any chance we can get Windows working here as well? It would be nice to move off of appveyor.

**Review by jdm — APPROVED** [review]
I'm definitely in favour of this after #28092 merges!

File: .github/workflows/rust.yml (hunk 1 above)
Comment (sagudev) [reply .github/workflows/rust.yml:15]: I was thinking the same. Will do.

**Comment by sagudev** [comment]
> 
> 
> I'm definitely in favour of this after #28092 merges!

#28092?

**Comment by CYBAI** [comment]
> #28092?

https://github.com/servo/servo/pull/28092

File: .github/workflows/rust.yml (hunk 1 above)
Comment (sagudev) [reply .github/workflows/rust.yml:15]:
```  DEBUG: pkg_config: Trying pkg-config
  INFO: c:/Strawberry/perl/bin/pkg-config
  INFO: checking for pkg-config version... 
  DEBUG: Executing: `c:/Strawberry/perl/bin/pkg-config --version`
  D:\a\mozjs\mozjs\makefile.cargo:190: recipe for target 'maybe-configure' failed

  --- stderr
  gmake: *** No targets specified and no makefile found.  Stop.
  mozmake: warning: -jN forced in submake: disabling jobserver mode.
  DEBUG: Executing: `c:/Strawberry/perl/bin/pkg-config --version`
  Traceback (most recent call last):
    File "D:\a\mozjs\mozjs\mozjs/js/src/../../configure.py", line 170, in <module>
      sys.exit(main(sys.argv))
    File "D:\a\mozjs\mozjs\mozjs/js/src/../../configure.py", line 46, in main
      sandbox.run(os.path.join(os.path.dirname(__file__), 'moz.configure'))
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 499, in run
      func(*args)
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 543, in _value_for
      return self._value_for_depends(obj)
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\util.py", line 997, in method_call
      cache[args] = self.func(instance, *args)
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 552, in _value_for_depends
      value = obj.result()
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\util.py", line 997, in method_call
      cache[args] = self.func(instance, *args)
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 155, in result
      return self._func(*resolved_args)
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 1152, in wrapped
      return new_func(*args, **kwargs)
    File "D:/a/mozjs/mozjs/mozjs/build/moz.configure/util.configure", line 433, in wrapper
      return func(*args)
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 1152, in wrapped
      return new_func(*args, **kwargs)
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 812, in wrapper
      ret = template(*args, **kwargs)
    File "D:/a/mozjs/mozjs/mozjs/build/moz.configure/checks.configure", line 58, in wrapped
      ret = func(*args, **kwargs)
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 1152, in wrapped
      return new_func(*args, **kwargs)
    File "D:/a/mozjs/mozjs/mozjs/build/moz.configure/pkg.configure", line 20, in pkg_config_version
      return Version(check_cmd_output(pkg_config, '--version').rstrip())
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 1152, in wrapped
      return new_func(*args, **kwargs)
    File "D:/a/mozjs/mozjs/mozjs/build/moz.configure/util.configure", line 55, in check_cmd_output
      retcode, stdout, stderr = get_cmd_output(*args, **kwargs)
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 1152, in wrapped
      return new_func(*args, **kwargs)
    File "D:/a/mozjs/mozjs/mozjs/build/moz.configure/util.configure", line 38, in get_cmd_output
      **kwargs)
    File "D:\a\mozjs\mozjs\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 902, in wrapper
      return function(*args, **kwargs)
    File "C:\hostedtoolcache\windows\Python\2.7.18\x64\Lib\subprocess.py", line 394, in __init__
      errread, errwrite)
    File "C:\hostedtoolcache\windows\Python\2.7.18\x64\Lib\subprocess.py", line 644, in _execute_child
      startupinfo)
  WindowsError: [Error 193] %1 is not a valid Win32 application
  mozmake: *** [maybe-configure] Error 1
  thread 'main' panicked at 'assertion failed: result.success()', build.rs:177:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[mozjs_sys 0.68.2] D:\a\mozjs\mozjs\makefile.cargo:190: recipe for target 'maybe-configure' failed
```

This is what I am getting in https://github.com/sagudev/mozjs/blob/win-actions/.github/workflows/rust.yml

Comment (sagudev) [reply .github/workflows/rust.yml]: Fixed!

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    - uses: actions/checkout@v2
+    - name: Install deps on osx
+      if: startsWith(matrix.os, 'macOS')
+      run: |
+        brew install python autoconf@2.13 ccache llvm yasm
+    - name: Install deps on linux
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        sudo apt install autoconf2.13 gcc-7 g++-7 ccache llvm -y
+    - uses: actions/setup-python@v2
```
Comment (jdm) [inline .github/workflows/rust.yml:31]: We can remove all uses of python2 from this workflow now.

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    - name: Build Windows
+      if: startsWith(matrix.os, 'windows')
+      shell: cmd
+      env:
+        MOZTOOLS_PATH: 'C:\mozilla-build\msys\bin;C:\mozilla-build\bin'
+        AUTOCONF: "C:/mozilla-build/msys/local/bin/autoconf-2.13"
+        LINKER: "lld-link.exe"
+        CC: "clang-cl.exe"
+        CXX: "clang-cl.exe"
+        NATIVE_WIN32_PYTHON: 'C:\hostedtoolcache\windows\Python\2.7.18\x64\python.exe'
```
Comment (jdm) [inline .github/workflows/rust.yml:72]: Let's make this python3 instead

**Review by jdm — CHANGES_REQUESTED** [review]
Looking good!

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -65,6 +65,7 @@ jobs:
         LINKER: "lld-link.exe"
         CC: "clang-cl.exe"
         CXX: "clang-cl.exe"
+        NATIVE_WIN32_PYTHON: "C:\\mozilla-build\\python2\\python.exe"
```
Comment (jdm) [inline .github/workflows/rust.yml:68]: Why is this required?

Comment (sagudev) [reply .github/workflows/rust.yml:68]: https://github.com/servo/mozjs/blob/master/makefile.cargo#L169

**Comment by jdm** [comment]
Filed https://github.com/servo/saltfs/issues/1013 to integrate the new workflow with homu. 21m end-to-end is incredible!

**Comment by sagudev** [comment]
Window builds are a little faster, macOS too, but the main improvement is direct async start of all builds.
slika
